BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -519,19 +546,25 @@ - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)re
 
 #pragma mark - BOXURLSessionTaskDelegate
 
-- (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response error:(NSError *)error
+- (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response responseData:(NSData *)responseData error:(NSError *)error
 {
-    [self finishURLSessionTaskWithData:self.responseData response:response error:error];
+    @synchronized (self) {
+        [self finishURLSessionTaskWithData:(responseData != nil ? responseData : self.responseData) response:response error:error];
+    }
 }
 
-- (void)processIntermediateResponse:(NSURLResponse *)response
+- (void)sessionTask:(NSURLSessionTask *)sessionTask processIntermediateResponse:(NSURLResponse *)response
 {
-    [self processResponse:response];
+    @synchronized (self) {
+        [self processResponse:response];
```

Comment:
added a FIXME to review if we need a check there. It's a bit more thought needed to make sure we treat nil as valid/invalid based on the type of operations